rotkehlchen/tasks/manager.py

```diff
@@ -550,10 +550,13 @@ def _maybe_update_snapshot_balances(self) -> Optional[list[gevent.Greenlet]]:
 
     def _maybe_query_produced_blocks(self) -> Optional[list[gevent.Greenlet]]:
         """Schedules the blocks production query if enough time has passed"""
+        if (beeaconchain := self.chains_aggregator.beaconchain).is_rate_limited():
+            return None
+
```

Comment:
this if check can be added to the one below, immediately after `self.chains_aggregator.get_module('eth2') is None or`